plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/MachineProviderImpl.java

```diff
@@ -518,7 +520,9 @@ protected void pullImage(CheServiceImpl service,
 
         try {
             boolean isSnapshot = SNAPSHOT_LOCATION_PATTERN.matcher(dockerMachineSource.getLocation()).matches();
-            if (!isSnapshot || snapshotUseRegistry) {
+            boolean isImageExistLocally = isDockerImageExistLocally(dockerMachineSource.getRepository());
+            if (!isSnapshot && (doForcePullImage || !isImageExistLocally) ||
```

Comment:
In my opinion, this code would be more clear if operands wouldn't rely on remembering operations priority. E.g. next equivalent code looks clearer for me:
```
(!isSnapshot && (doForcePullImage || !isImageExistLocally) || isSnapshot && snapshotUseRegistry)
```